homeassistant/components/sensor/hddtemp.py

```diff
@@ -84,17 +81,18 @@ def state(self):
     @property
     def unit_of_measurement(self):
         """Return the unit the value is expressed in."""
-        if self._details[3] == 'C':
-            return TEMP_CELSIUS
-        return TEMP_FAHRENHEIT
+        if self._details is not None and self._details[3] == 'F':
+            return TEMP_FAHRENHEIT
+        return TEMP_CELSIUS
```

Comment:
Let's save the object as `self._unit`, and move this logic into `update` after assigning `self._state`.

That way if the sensor isn't available on startup, the units will still be initialized as `None`, and if the sensor becomes unreachable later, it will remember it's units while offline, since it shouldn't change. You could alternately clear it out if you wanted, but that seems unnecessary.

Reply:
@armills Thank you for the feedback. I moved that into `update` as you suggested.